### PR TITLE
Revert "Keystrokes to adjust applications volume and mute"

### LINKED ISSUE #16402: Unmuting other apps does not work as expected
### Steps to reproduce:
1. Launch NVDA and an audio stream in the background, e.g. youtube playing music in  a browser
2. Press NVDA+alt+delete to mute all apps except NVDA.
As expected, Youtube is muted.
3. Exit NVDA
4. Start NVDA again
5. Press NVDA+alt+delete
6. Press NVDA+alt+delete

### Actual behavior:
2. Youtube stream is muted
3. Youtube stream remains muted
4. Youtube stream remains muted
5. NVDA says "Applications muted" and Youtube stream remains muted
6. NVDA says "Applications unmuted" and Youtube stream is restored (unmuted)

### Expected behavior:
At least, at step 5, NVDA should already know that the other apps are muted and it should unmute them.

I really think that NVDA should not store in its config nor manage an additional muted state. For clarity it should be able to mute and unmute directly the sound as configured in Windows Volume mixer. Else people may have muted through the volume mixer and can wonder why they cannot unmute through NVDA. In other words, putting two switches serially leads to a confusing user experience.

I have not tested but we may encounter the same issues with other apps volume management commands.

In #16273, you indicate that you are using the same framework for mute other apps or other apps volume commands than in sound split. IMO muted state and volume of ther apps should not be stored in NVDA's config, nor should they be restored or re-configured at NVDA exit/startup. They should just be a more easy way to control Windows Mixer's parameters. On the other side, Split mode should be set up according to NVDA's config and disabled when NVDA exits since it's a feature that only has sense when NVDA is active; thus it makes sense to manage it internally to NVDA and Windows Volume Mixer should not be touched for Sound Split feature.

### NVDA logs, crash dumps and other attachments:
None
#### NVDA installed/portable/running from source:
From source
#### NVDA version:
Last alpha, commit 3d4a0a8f7b1d124231e9774d30d07cc549dac3fc.
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.4170
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Not tested.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No; new feature
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tested

### LINKED ISSUE #16409: Apps mute and volume features work very unexpectedly with WASAPI disabled
### Steps to reproduce:

1. IN NVDA Advanced settings, set WASAPI to disabled.
2. Apply, save config, and restart NVDA.
3. Start some sound playing. I used Firefox, playing a video.
4. Attempt to mute other apps (`NVDA+Alt+Del`).
5. Attempt to change volume of other apps (`NVDA+Alt+pgDn`).
6. Continue lowering volume of other apps to 20 or so.
7. In NVDA Advanced settings, enable WASAPI.
8. Save config and restart NVDA.

#### At step 4, NVDA says:

>Sound split cannot be used. Please enable WASAPI in the Advanced category in NVDA Settings to use it.
>Applications muted

First,. I was not trying to use Sound Split. Why is it being mentioned?
Second, Applications have not been muted, so the message is false.

#### At step 5, NVDA says:

>Sound split cannot be used. Please enable WASAPI in the Advanced category in NVDA Settings to use it.
>Applications volume 95

First, I was not using Sound Split, again.
Second, the application volume was not changed, so the message is false.

#### At step 8:

Upon NVDA restart, the volume of the other application--which was previously not changed--suddenly drops precipitously to match the volume that was set while WASAPI was disabled.

### Expected behavior:

At step 4, NVDA should speak only something like:

>Application mute is only supported when WASAPI is enabled in NVDA advanced settings.

At step 5, NVDA should speak only something like:

>Application volume changes are only supported when WASAPI is enabled in NVDA advanced settings.

At step 8, the volume should not change on NVDA startup, because step 4 should not have changed any volume in configuration.

#### NVDA installed/portable/running from source:

Portable.

#### NVDA version:
alpha-31622,11a2b4f1

#### Windows version:
11 HOME 23H3

#### Name and version of other software in use when reproducing the issue:
N/A

#### Does the issue still occur after restarting your computer?
N/A

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Of course.

### PR DESCRIPTION
**Pull request by seanbudd**
### Reverts PR
Reverts nvaccess/nvda#16273

### Issues fixed
Fixes #16409 
Fixes #16402 

### Issues reopened
#16052 

### Brief reason for revert

We started with mltony creating:
#16051 Feature request: Sound split

Which was a duplicate of:
#12985 Audio settings with stereo headset (or speakers) - Send NVDA sounds on one side and the rest of Windows sounds on the other side

And then implemented by:
#16071 Sound split

mltony also created:
#16052 Feature request: add command to adjust volume of all applications except for NVDA

Which was implemented in:
#16273 Keystrokes to adjust applications volume and mute

This PR was approved and merged and was then found to cause issues:
#16402 Unmuting other apps does not work as expected
#16409 Apps mute and volume features work very unexpectedly with WASAPI disabled

Due to these issues and the considerable debate on the approach, the above PR #16273 was reverted by:
#16440

As an alternative to the revert #16440 to resolve the 2 issues (#16402, #16409) and keep #16273, mltony created:
#16404

The question now becomes, how do we proceed from here? 

NV Access's position is that the sound split functionality (#16051) is a useful feature to add into core. However, due to the following reasons, we believe that further work on the volume adjustment features (#16273, #16404) to improve the UX is required on a branch (off master/alpha) before it can be added back in:
- Windows sound mixer has reasonable accessibility.
- Sound split on its own provides value to users.
- The UX of swapping between NVDA volume control and windows volume control needs to be resolved.
- The UX of resolving volume issues due to NVDA crashes needs to be resolved. 

As one of the contributors on the threads said, "So now there are two mixers in the chain, one of which can be invisible, and overrides the other, or makes its settings relative instead of absolute."

### CODE REVIEW

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -50,10 +50,8 @@ There are many minor bug fixes for applications, such as Thunderbird, Adobe Read
     -
   -
 - Sound split: (#12985, @mltony)
-  - Allows splitting NVDA sounds into one channel (e.g. left) while sounds from all other applications are placed in the other channel (e.g. right).
+  - Allows splitting NVDA sounds in one channel (e.g. left) while sounds from all other applications in the other channel (e.g. right).
```
Comment (gerald-hartig) [inline user_docs/en/changes.t2t:53]: Although this is what we had before, the PR being reverted fixed the phrasing. Can we skip reverting this bit?

**Review by gerald-hartig — DISMISSED** [review]

**Comment by Adriani90** [comment]
@seanbudd in the expectation that PR templates should be filled properly, could you please take the example role you have on the community and explain the reasons?
In #16404 Tony invested time and ressources to fix the problems reported, please explain why that is not accepted.
cc: @jcsteh, @michaelDCurran, @LeonarddeR.

**Comment by gerald-hartig** [comment]
@Adriani90 @mltony I've updated the description of the revert PR to make our thinking a bit more explicit. If you (or anyone else in the community) would like to continue the conversation please feel free to open a discussion thread. Your hard work and dedication to improving NVDA is greatly appreciated and what makes this community so valuable.

**Comment by mltony** [comment]
@gerald-hartig, @seanbudd 
To make sure my understanding is correct, can you confirm that you'd accept the next version of PR if:
1. Application volume when adjusted by NVDA keystrokes is reflected in Windows Mixer.
2. Application volume and mute commands work regardless sound split being on or off.
3. Application volume and mute status is restore upon exit.
4. Application volume, but not mute status is persistent across NVDA restarts. I believe in the previous discussion someone insisted on not muting apps at startup even if they were muted in the previous session.
5. When apps volume is 100% and mute is off, then this feature won't make a single wasapi call in order to let third party add-ons or applications to work correctly.
Am I missing any other concerns? It would be good to agree on precise list of requirements since this feature provokes so lengthy discussions.

**Comment by gerald-hartig** [comment]
@mltony We'd be happy with the PR if it addressed those 5 things.

**Comment by Adriani90** [comment]
@gerald-hartig, @mltony I am struggling trying to get why the bar is set higher for this feature in one aspect:
> 1. Application volume when adjusted by NVDA keystrokes is reflected in Windows Mixer.

This is not what we have in NVDA. Try as follows:
1. Set audio ducking on an app to always via nvda+shift+d key (e.g. in Firefox)
2. Start a youtube video
3. Use NVDA to navigate so that the volume of the youtube video is lowered
4. Check the volume mixer to see the volume in firefox.

Currently, the volume mixer does not reflect the volume adjustments done by the audio ducking feature.

I think it really makes sense to accept this feature only if it is an NVDA internal change, without any reflection in the Windows mixer.
I think then the implementation is less complex if this is NVDA internal only, and does not depend on any major design changes to the volume mixer which Microsoft might plan in the future.

**Comment by CyrilleB79** [comment]
@gerald-hartig, @seanbudd, as you wrote, there has been a "considerable debate on the approach".

@mltony has asked a confirmation about 5 design choices in https://github.com/nvaccess/nvda/pull/16440#issuecomment-2083893274.

> 1. Application volume when adjusted by NVDA keystrokes is reflected in Windows Mixer.
> 2. Application volume and mute commands work regardless sound split being on or off.
> 3. Application volume and mute status is restore upon exit.
> 4. Application volume, but not mute status is persistent across NVDA restarts. I believe in the previous discussion someone insisted on not muting apps at startup even if they were muted in the previous session.
> 5. When apps volume is 100% and mute is off, then this feature won't make a single wasapi call in order to let third party add-ons or applications to work correctly.

For point 1, I would add that not only apps volumes but also mute state should be reflected in the Windows Mixer.
And I disagree with point 3 and 4 as explained here and there in the various discussions; changing the parameters in Windows Mixer when NVDA starts up or exit is quite a bad UX IMO and moreover, having a different behaviour for mute and apps volume at NVDA startup would make things more confusing.
Should I expand on this here again or elsewhere?

I have the feeling that our discussions on this topic is quite inefficient because it is scattered in many GitHub issues and PRs. Before going again forward with a PR, we really should have a centralized discussion on this feature with designed choices not only agreed by NV Access but also explained.

Among others, the following questions on the mute and apps volume change feature should be answered with justification:
1. Is this feature (apps volume and mute) required for SoundSplit to be useful?
2. Should the feature impact Windows Volume Mixer (volume sliders and mute buttons) or not?
3. Should NVDA be allowed to control Windows Volume Mixer on startup / exit or only when keystrokes are used?
4. Do we accept or not that volume and mute commands do not act at the same level, e.g. internal muting in NVDA but volume changes reflected in the Mixer?
5. If the commands act on Windows Volume Mixer, are the controls in NVDA's audio panel still needed?
6. Is the feature working as expected even in case of NVDA crash? If not is there a workaround?

I won't answer here for myself before we know where all this should be discussed but will do as soon as the discussion location is clarified. It would be nice that either @mltony or NV Access indicate where this discussion should take place and initiate it because you are more legitimate. But if no one does, I may do it myself.

**Comment by Adriani90** [comment]
@CyrilleB79 
> For point 1, I would add that not only apps volumes but also mute state should be reflected in the Windows Mixer.

You seem to totally ignore my comments. This approach is not what we have in NVDA currently. So if NV Access decides to reflect the changes in the windows mixer, then this should be changed for audio ducking as well.
But then we introduce two major risks:
1. More complexity in the implementation and
2. NVDA will depend stronger on designs of volume mixer which might change in the future, resulting in this feature to fail.
So I really advise to not reflect these changes in the windows mixer.

> And I disagree with point 3 and 4 as explained here and there in the various discussions; changing the parameters in Windows Mixer when NVDA starts up or exit is quite a bad UX IMO

Again, @CyrilleB79  and also @mltony  you totally ignored my comments. Neither apps volume nor mute / unmute state should be reflected in the mixer. All of them should be restored after NVDA exit and should remain a NVDA internal change. That's what we have since years with all audio related features, e.g. audio ducking.
You also ignore the use case of people with hearing problems using the machine together with visually impaired people.

> 1. Is this feature (apps volume and mute) required for SoundSplit to be useful?

Offcourse it is useful, at least as useful as audio ducking.

> 2. Should the feature impact Windows Volume Mixer (volume sliders and mute buttons) or not?

In my opinion these should never touch the volume mixer at all.

> 3. Should NVDA be allowed to control Windows Volume Mixer on startup / exit or only when keystrokes are used?

Vomume mixer should not be involved at all.

And questions 4-6 are obsolete in this case.

**Comment by LeonarddeR** [comment]
For what its worth, I completely second the expectations as raised by @gerald-hartig. At a certain point we really need to respect choices made by NV Access, without continuing to engage in endless discussions.

@Adriani90 Comparing the audio ducking feature with lowering app volume is the same as comparing apples and oranges. Audio ducking is a feature which is completely separate from the volume mixer, namely a Windows accessibility feature. More importantly though, the ducker feature is bound to the application window and even when NVDA is forcefully killed with the taskkill command, ducking is disabled.

If NVDA fiddles with the volume of any application, it should do so in a standardized way that is reflected by the volume mixer. If not, a sighted user is unable to fix volume changes after a force shutdown of NVDA.

**Comment by CyrilleB79** [comment]
@Adriani90, I have read your comments but I still disagree with it.
However, as explained in my previous comment, I am not sure that this PR is the right location to discuss it; that's why I do not expand here on why I disagree with you.

**Comment by Adriani90** [comment]
> Comparing the audio ducking feature with lowering app volume is the same as comparing apples and oranges. Audio ducking is a feature which is completely separate from the volume mixer, namely a Windows accessibility feature. More importantly though, the ducker feature is bound to the application window and even when NVDA is forcefully killed with the taskkill command, ducking is disabled.

Thanks for this qualified answer, very appreciated. I didn't know audio ducking is a native windows accessibility feature  separated from all other volume related things. I thought NVDA found a way to avoide the volume mixer through other internal adjustments. Sometimes a qualified technical answer to some comments would really reduce the amount of noise in such discussions and people like me would feel more seriously taken.

File: user_docs/en/userGuide.t2t [hunk 2]
```diff
@@ -1998,15 +1998,11 @@ By default this command will cycle between the following modes:
...
 
 There are more advanced sound split modes available in NVDA setting combo box.
-If you wish to adjust volume of all applications except for NVDA, consider using [the dedicated commands #OtherAppVolume].
 Please note, that sound split doesn't work as a mixer.
 For example, if an application is playing a stereo sound track while sound split is set to "NVDA on the left and applications on the right", then you will only hear the right channel of the sound track, while the left channel of the sound track will be muted.
 
 This option is not available if you have started NVDA with [WASAPI disabled for audio output #WASAPI] in Advanced Settings.
 
-Please note, that if NVDA crashes, then it won't be able to restore application sounds volume, and those applications might still output sound only in one channel after NVDA crash.
-In order to mitigate this, please restart NVDA.
```
Comment (seanbudd) [inline user_docs/en/userGuide.t2t:2008]: @mltony - should this section also be kept - i.e. is the issue surrounding sound split being stuck after a crash still an issue?

File: user_docs/en/changes.t2t (hunk 1 above)
Comment (seanbudd) [reply user_docs/en/changes.t2t:53]:
```suggestion
  - Allows splitting NVDA sounds into one channel (e.g. left) while sounds from all other applications are directed to the other channel (e.g. right).
```

File: user_docs/en/userGuide.t2t (hunk 2 above)
Comment (mltony) [reply user_docs/en/userGuide.t2t:2008]: Yes, this section should be kept. There is nothing we can do to restore volumes of apps when NVDA crashes.

**Review by Qchristensen — DISMISSED** [review]
User Guide changes (reverts) are fine.

**Comment by Adriani90** [comment]
@LeonarddeR Tony wrote:
> Yes, this section should be kept. There is nothing we can do to restore volumes of apps when NVDA crashes.

This is revering to sound split feature.
How is it with audio cucking set to !always"? Can we be sure that NVDA restores the volume after exit? Even if it is an accessiility feature in Windows, it is still not reflected in the volume mixer. So if audio ducking is not able to restore volume after NVDA crash, we would ent up with the same problem you describe, that means that sighted people would not be able to help in restoring the volume of apps. In this case it might not be that bad because audio ducking is not lowering the volume that much, but in theory we have the same problem there I guess.

**Pull request merged**